### PR TITLE
Refactor: Buffer

### PR DESCRIPTION
**Pull request by cainthebest**
Ref: #27 
Fix's: #26

### CODE REVIEW

File: src/bufferer.rs [hunk 1]
```diff
@@ -1,312 +1,292 @@
...
+        let delimiter = until.unwrap_or(D::DELIMITER);
+
+        let position = match self.data[self.cursor ..]
+            .windows(delimiter.as_ref().len())
+            .position(|window| window == delimiter.as_ref())
+        {
+            Some(pos) => pos,
+            None => {
+                if optional {
+                    self.data.len() - self.cursor
```
[the remark this answers is not in the document]
Comment (cainthebest) [reply src/bufferer.rs:110]: I literally have a commit that im about to push that changes things around as i has a weird utf16 bug some reason

**Comment by CosminPerRam** [comment]
This is a very impressive rework!
After a glance it looks like this is ready to replace the old bufferer, whats your thoughts on this? If there isn't much to say I could start working on it (as we discussed over the Discord server that I could make the transition).

**Comment by cainthebest** [comment]
Yeah it looks good to me as far as i have looked, Im happy to handover the PR to you

**Comment by cainthebest** [comment]
@CosminPerRam 1 thing though, do we keep with `bufferer.rs` or rename to `buffer.rs. Its only internally used

**Comment by CosminPerRam** [comment]
We could rename it to buffer.

**Pull request merged**